src/backend/strategies.rs

```diff
@@ -1246,262 +1531,177 @@ pub async fn run_strategy_task<'s>(
+                                                                _ => panic!("This should never happen")
                                                             }
-                                                        } else {
-                                                            tracing::info!("Response version other than V0 received or absent for state transition {} ({})", st_queue_index, transition_type);
-                                                        }
+                                                        ).collect_vec(),
+                                                        _ => panic!("This shouldn't happen")
+                                                    };
+                                                    for contract_id in contract_ids {
+                                                        let mut mempool_document_counter_clone_lock = mempool_document_counter_clone.lock().await;
+                                                        let count = mempool_document_counter_clone_lock.entry((transition_clone.owner_id(), contract_id)).or_insert(0);
+                                                        *count += 1;
+                                                        // tracing::info!(" + Incremented identity {} tx counter for contract {}. Count: {}", transition_owner_id, contract_id.to_string(Encoding::Base58), count);
                                                     }
-                                                    Err(e) => tracing::error!(
-                                                        "Error waiting for state transition result: {:?}",
-                                                        e
-                                                    ),
                                                 }
-                                            } else {
-                                                tracing::error!(
-                                                    "Failed to create wait request for state transition."
-                                                );
+                                                Ok((transition_clone, broadcast_result))
+                                            },
+                                            Err(e) => {
+                                                errs.fetch_add(1, Ordering::SeqCst);
+                                                tracing::error!("Error: Failed to broadcast {} transition: {:?}. ID: {}", transition_clone.name(), e, transition_id);
+                                                if e.to_string().contains("Insufficient identity") {
+                                                    insufficient_balance_error_count += 1;
+                                                    // Top up. This logic works but it slows the broadcasting down slightly.
```

Comment:
not needed anymore maybe?

Reply:
I activated it and made it optional to the user